### ISSUE TITLE
Docker Desktop doesn't respect integration with other WSL 2 distros

### ISSUE THREAD

**Issue by PavelSosin-320** [docker/for-win#6894]
According to **Error response from daemon: mkdir .. permission denied** message issued when mounting bind volumes with a source inside another WSL 2 distro FS Docker Desktop doesn't respect integration with other WSL 2 doesn't respect the corresponding configuration.

  - Windows Version: Win 10 Pro 2005
  - Docker Desktop Version: 2.3.1
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior

  1. Install Ubuntu distro in Windows 10 
  2.  Enable WSL feature
  3. Enable WSL distro integration with Ubuntu
  4. Run either docker run or Docker service create with volume mount like  --mount type=bind,source=git,target=/var/git from Ubuntu distro terminal.
message Error response from daemon: mkdir git: permission denied appears.
It indicates that docker daemon access Ubuntu as a user with insufficient authorization.
-------
WSL 2 architecture assumes that 2 distro s, i.e. lightweight VMs interact with respect to login user like in wsl -u <username>  to gain access to Linux resources.
Anonymous access is insufficient.
Integration with another distro shall be in the form user:password@distro where user default can be root to avoid complexity. Otherwise, login user must have sudo authorization by default.
According to my test root access w/o password is possible:
```
C:\Users\Pavel>wsl -d Ubuntu -u root whoami
root
but not desired due to lack of user-specific profile
```
Diagnostics ID : 1695A9B2-5AAE-4023-953F-F08D7698FC5A/20200527052236 no usable info

**Comment by PavelSosin-320** [comment]
Shortly, instead of mkdir daemon shall use sudo mkdir -m -p

**Comment by simonferquel** [comment]
When mounting a non-existing directory, mkdir is run from the desktop-proxy which runs as root within your distro. So I don't think your issue is related to that. I'll have a look at your logs.

**Comment by simonferquel** [comment]
Ok, nailed it:
- bind mount sources must be absolute paths ("git" is not an absolute path)
- the desktop proxy running in your distro incorrectly accept non-absolute paths and try to create relative dirs to the desktop-proxy working dir (which is /mnt/c/Program Files/Docker/Docker). 
- the desktop proxy runs as root but access Windows files with the current user access rights which does not have rights to write in `C:\Program Files\Docker\Docker`). Thus it ends with an access denied error.

The error should be different, asking the user for an absolute path. I'll add it to our backlog.

**Comment by mat007** [comment]
This issue has been fixed in Edge 2.3.6.0, I’ll go ahead and close this ticket.
Thanks!

**Comment by djs55** [comment]
The fix from Edge 2.3.6.0 has now also been released in [Stable 2.4.0.0](https://desktop.docker.com/win/stable/48506/Docker%20Desktop%20Installer.exe).
